### ISSUE TITLE
perf: Delay / FOUC after transitioning between views

### ISSUE THREAD

**Issue by olliebennett** [driftyco/ionic#3495]
**Type**: <span ionic-type>perf</span>

**Platform**: <span ionic-platform>ios</span> <span ionic-platform-version>8</span>  <span ionic-webview>webview</span>

<span ionic-description>Since Ionic v1.0.0-beta.14-nightly-993 ([diff](https://github.com/driftyco/ionic-bower/compare/v1.0.0-beta.14-nightly-992...v1.0.0-beta.14-nightly-993)), we have identified very slow hiding of `ng-hide` elements in the newly transitioned view.

This is apparent on an iOS device (not emulator) and not on Android devices. Tested with iPhone 6+. Prior to this release, there was only a very minor (acceptable) flicker during the transition.

The extent of the delay is reduced by avoiding excessive usage of `ng-hide`, but is still substantially worse than the previous (`v1.0.0-beta.14-nightly-992`) behaviour.

Assuming the regression was due to the iPhone 'swipe back' feature ([this commit](https://github.com/driftyco/ionic/commit/8ebde73d0b8afac1bf1c1787c90a72a28a88bc3a) is the main contributor to the diff linked above), we set `$ionicConfigProvider.views.swipeBackEnabled(false);` without effect.

Disabling view caching (`$ionicConfigProvider.views.maxCache(0);`) has no effect on the initial view load (but obviously improves performance of subsequent views).

Our solution is to revert to `v1.0.0-beta.14` for now. Reducing `ng-hide` usage (in favour of `ng-if` where appropriate, for example) helps but does not resolve. Updating to latest angular/ionic nightlies also had no noticeable effect.

Dependencies in our `bower.json`:
```
"angular": "1.3.6",
"angular-resource": "1.3.6",
"ionic": "driftyco/ionic-bower#1.0.0-beta.14-nightly-993",
"ngCordova": "~0.1.14-alpha",
"ionicons": "~2.0.1"
```

Plugin list:
```
com.danielcwilson.plugins.googleanalytics 0.6.1 "Google Universal Analytics Plugin"
com.ionic.keyboard 1.0.4 "Keyboard"
com.phonegap.plugins.facebookconnect 0.11.0 "Facebook Connect"
nl.x-services.plugins.socialsharing 4.3.15 "SocialSharing"
org.apache.cordova.console 0.2.13 "Console"
org.apache.cordova.device 0.3.0 "Device"
org.apache.cordova.inappbrowser 0.6.0 "InAppBrowser"
org.apache.cordova.splashscreen 1.0.0 "Splashscreen"
org.apache.cordova.statusbar 0.1.10 "StatusBar"
uk.co.whiteoctober.cordova.appversion 0.1.5 "AppVersion"
```

</span>

<span is-issue-template></span>

**Comment by mhartington** [comment]
Could you provide a codepen demo that we can test out?

**Comment by AlexYankee** [comment]
This is probably can be related to ngAnimate

**Comment by jtomaszewski** [comment]
You can be interested in this issue and workarounds - https://github.com/angular/angular.js/issues/8224 .

My workaround which works:
```scss
// Append .no-ng-animate to an object with [ng-show] / [ng-hide] which has an infinite animation
// This fixes show-hide delay bug on f.e. spinner elements
// More information: https://github.com/angular/angular.js/issues/8224
.no-ng-animate {
  &.ng-animate {
    @include animation(none 0s);
  }

  &.ng-hide-add {
    transition: 0s linear all;
  }
}
```

**Comment by jtomaszewski** [comment]
Workaround:
```scss
// Fix for ng-show/ng-hide issue on Angular 1.3 & Ionic >1.0.0beta14
// (http://forum.ionicframework.com/t/beta-14-ng-hide-show/14270/8)
.ng-hide-add,
.ng-hide-remove {
  display: none !important;
}
```

**Comment by mhartington** [comment]
Alright, yeah this is related to ng-animate and not ionic core.
